impeller/display_list/canvas.h

```diff
@@ -24,10 +25,19 @@
 #include "impeller/geometry/path.h"
 #include "impeller/geometry/point.h"
 #include "impeller/geometry/vector.h"
+#include "impeller/renderer/snapshot.h"
 #include "impeller/typographer/text_frame.h"
 
 namespace impeller {
 
+struct BackdropData {
+  size_t backdrop_count = 0;
+  bool all_filters_equal = true;
+  std::shared_ptr<Texture> texture_slot;
+  std::optional<Snapshot> filtered_input_slot;
+  const flutter::DlImageFilter* last_backdrop = nullptr;
```

Comment:
maybe we can just do this.

Reply:
> hash is not equality though.

It can be if your hashes are guaranteed to be unique (to a precision you care about).  Probably better to pursue dispatching something else though like you mention.